juicebox_scripts/juicebox_assembly_converter.py

```diff
@@ -190,7 +190,7 @@ def _read_assembly(self, assembly, contig_mode=False):
                         raise MissingFragmentError('Assembly {0} is missing the sequence for index {1}'.format(assembly, len(assembly_map) + 1))
                     if int(tokens[2]) == 0:
                         raise ZeroLengthContigError('Assembly {0} lists contig {1} as zero length'.format(assembly, tokens[0]))
-                    assembly_map.append((tokens[0], tokens[2]))
```

Comment:
i feel like the hardcoded n-1, n+1 game is how we got into trouble, but i don't have a better simple idea for fixing.

Reply:
Yep, but it's required because of different file specs. The length token in a .assembly file is equal to the length of the sequence + 1. We're changing it here so we at least encode the real length in our internal data structure, and use the output functions to twiddle things around when necessary for a particular file format. The updated test cases also are related to this -- we thought the length token was equal to length, not length + 1 before